### PR TITLE
Auto detect server name for packaged servers.

### LINKED ISSUE #83: Determine packaged server name automatically
When using a packaged server with a non-default server name (i.e. `defaultServer`), it is required that `BP_OPEN_LIBERTY_SERVER_NAME` be set. If there is only one possible server to use and `BP_OPEN_LIBERTY_SERVER_NAME` is not set, we should automatically detect the server name.

Also, we may need to improve the error here also if a packaged server was provided but server was not found as it was not obvious when someone was not using the correct server name.

### PR DESCRIPTION
**Pull request by hibell**
## Summary
Auto detect server name for packaged servers. Also fixes an issue where the server name was not being found by the linker
which resulted in the defaultServer being used for server name regardless of whatever value `BP_LIBERTY_SERVER_NAME` was set to.

Fixes #83.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Review by dmikusa-pivotal — APPROVED** [review]

File: buildpack.toml [hunk 1]
```diff
@@ -75,8 +75,8 @@ api = "0.7"
 
   [[metadata.configurations]]
     build = true
-    default = "defaultServer"
-    description = "Name of the server to use. Defaults to defaultServer"
+    default = ""
+    description = "Name of the server to use. If not set, the buildpack tries to detect the server name for packaged servers. Otherwise, it defaults to 'defaultServer'"
```
[the remark this answers is not in the document]
Comment (hibell) [reply buildpack.toml:79]: That sounds good.

File: liberty/build.go [hunk 2]
```diff
@@ -172,22 +175,18 @@ func createStackRuntimeProcess(serverName string) (libcnb.Process, error) {
 }
 
 func (b Build) checkJvmApplicationProvided(context libcnb.BuildContext, serverName string) (bool, error) {
-	isPackagedServer := isPackagedServerPlan(context.Plan.Entries)
+	usrPath, isPackagedServer := getPlanMetadata("packaged-server-usr-path", context.Plan.Entries)
```
[the remark this answers is not in the document]
Comment (hibell) [reply liberty/build.go]: I thought using the build plan was the preferred mechanism for passing information between the detect and build phases per the CNB doc:
> The Build Plan is a document the buildpacks can use to pass information between the detect and build phases. The build plan is passed (by the lifecycle) as a parameter to the detect and build binaries of the buildpack.

But, as you said, it is a bit cumbersome and hard to follow. I refactored the detect an app validation logic and just reuse in detect, build, and linker now.

**Review by dmikusa-pivotal — APPROVED** [review]
